### PR TITLE
Allow --creds flag

### PR DESCRIPTION
**Pull request by derekcollison**
I tried several versions of this, feels like it should be easier.

First attempt always called with_credentials() and wanted it to do right thing with empty creds.

Second attempt built up options, but when I tried top reassign with credentials complained because it was different types.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: examples/nats-box/main.rs [hunk 1]
```diff
@@ -28,12 +31,24 @@ enum Command {
 
 fn main() -> CliResult {
     let args = Cli::from_args();
-    let nc = nats::ConnectionOptions::new()
-        .with_name("nats-box rust example")
-        .connect(&args.server)?;
+    let nc: nats::Connection;
+
+    if args.creds.len() > 0 {
```
Comment (spacejam) [inline examples/nats-box/main.rs:36]: can creds be an `Option<String>` to avoid conflating emptiness with presence?

Comment (derekcollison) [reply examples/nats-box/main.rs:36]: Not sure, depends on CLI framework.

Regardless of that, what would your optimum connection constructor look like here with optional creds?

Comment (spacejam) [reply examples/nats-box/main.rs:36]: I'd recommend doing this: https://github.com/nats-io/nats.rs/blob/3c8e91690d337f3cbe499e5a48a46288216cae0d/examples/nats_bench.rs#L19-L21

Comment (derekcollison) [reply examples/nats-box/main.rs:36]: That is why I asked, I like it! I will make the changes.. Thanks!

Comment (derekcollison) [reply examples/nats-box/main.rs]: Please take another look, thanks again for the suggestion, exactly what I was looking for.

**Pull request merged**